docs/_docs/continuous-integration/github-actions.md

```diff
@@ -0,0 +1,208 @@
+
+The Jekyll site we'll be using for the rest of this page initially consists of just a `_config.yml`,
+an `index.md` page and a `Gemfile`. The contents are respectively:
+
+```yaml
+# _config.yml
+
+title: "Jekyll Actions Demo"
+```
+
+{% raw %}
+```liquid
+---
+---
+
+Welcome to My Home Page
+
+{% assign date = '2020-04-13T10:20:00Z' %}
+
+- Original date - {{ date }}
+- With timeago filter - {{ date | timeago }}
+```
+{% endraw %}
+
+
+```ruby
+# Gemfile
+
+source 'https://rubygems.org'
+
+gem 'jekyll', '~> 4.0'
+
+group :jekyll_plugins do
+  gem 'jekyll-timeago', '~> 0.13.1'
+end
+```
+
+{: .note .info}
+The demo site uses Jekyll 4 and a [third-party plugin][timeago-plugin], both of which are currently
+not whitelisted for use on GitHub pages. The plugin will allow us to describe how far back a date
+was from today. e.g. If we give a date as `2016-03-23T10:20:00Z` and the current date is
+`2020-04-13T10:20:00Z`, then the output would be `4 years and 3 weeks ago`.
+
+{: .note .info}
+The action we're using takes care of installing the Ruby gems and dependencies. While that keeps
+the setup simple for the user, one may encounter issues if they also check-in `Gemfile.lock` if it
+was generated with an old version of Bundler.
+
+### Setting up the Action
+
+GitHub Actions are registered for a repository by using a YAML file inside the directory path
+`.github/worksflows` (note the dot at the start). Here we shall employ
+[Jekyll Actions][jekyll-actions] from the Marketplace for its simplicity.
+
+Create a **workflow file**, say `jekyll.yml`, using either the GitHub interface or by pushing
+a YAML file to directory path manually. The base contents are:
+
+{% raw %}
+```yaml
+name: Build and deploy Jekyll site to GitHub Pages
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  jekyll:
+    runs-on: ubuntu-16.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: helaili/jekyll-action@2.0.0
+        env:
+          JEKYLL_PAT: ${{ secrets.JEKYLL_PAT }}
+```
+{% endraw %}
+
+The above workflow can be explained as the following:
+
+- We trigger the build on a **push** to `master` branch only --- this prevents the Action from
+  overwriting the `gh-pages` branch on feature branch pushes.
+- The name of the job matches our YAML filename: `jekyll`.
+- The **checkout** action takes care of cloning your repository.
+- We specify our selected **action** and **version number** using `helaili/jekyll-action@2.0.0`.
+  This handles the build and deploy.
+- We set a reference to a secret **environment variable** for the action to use. The `JEKYLL_PAT`
+  is a *Personal Access Token* and is detailed in the next section.
+
+As an alternative to building on a push event, you can trigger a build using the [on.schedule]
+field. For example, to build nightly at midnight:
+
+```yaml
+on:
+  schedule:
+    - cron:  '0 0 * * *'
+```
+
+[on.schedule]: https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onschedule
+
```

Comment:
Most people don't know the `cron` syntax, a link to the  https://crontab.guru/ help would be a nice addition here.